packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -1019,6 +1020,14 @@ export class UserPool extends UserPoolBase {
 }
 
 function undefinedIfNoKeys(struct: object): object | undefined {
-  const allUndefined = Object.values(struct).reduce((acc, v) => acc && (v === undefined), true);
+  const allUndefined = Object.values(struct).every(val => val === undefined);
   return allUndefined ? undefined : struct;
 }
+function encodeDomain(email: string | undefined): string | undefined {
+  const splitEmail = email?.split('@');
+  if (splitEmail === undefined || splitEmail.length !== 2) {
+    return email;
+  }
+  const [local, domain] = splitEmail!;
+  return `${local}@${punycodeEncode(domain)}`;
+}
```

Comment:
Can't we just use the [`toASCII()`](https://www.npmjs.com/package/punycode#punycodetoasciiinput) API instead. According to the documentation, it takes care of emails.

Reply:
Yeah, but it also encodes the part before the `@`, which for AWS/Cloudformation should not be encoded. See rix0rr's earlier comment and the [documentation](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/DomainNameFormat.html).

Reply:
Not according to the documented example that I've linked above. Specifically example#3 shows that the first half of the email address is *NOT* encoded.

Does the library behave differently? Can you test this out locally?

Reply:
You are correct! Did not go through the documentation properly!

I reverted back to the `toASCII` method, as previously.